### PR TITLE
port define-prefix-command

### LINKED ISSUE #410: Port define-prefix-command to Rust
```c
DEFUN ("define-prefix-command", Fdefine_prefix_command, Sdefine_prefix_command, 1, 3, 0,
       doc: /* Define COMMAND as a prefix command.  COMMAND should be a symbol.
A new sparse keymap is stored as COMMAND's function definition and its
value.
This prepares COMMAND for use as a prefix key's binding.
If a second optional argument MAPVAR is given, it should be a symbol.
The map is then stored as MAPVAR's value instead of as COMMAND's
value; but COMMAND is still defined as a function.
The third optional argument NAME, if given, supplies a menu name
string for the map.  This is required to use the keymap as a menu.
This function returns COMMAND.  */)
  (Lisp_Object command, Lisp_Object mapvar, Lisp_Object name)
{
  Lisp_Object map;
  map = Fmake_sparse_keymap (name);
  Ffset (command, map);
  if (!NILP (mapvar))
    Fset (mapvar, map);
  else
    Fset (command, map);
  return command;
}
```

### PR DESCRIPTION
**Pull request by brotzeit**
closes  #410

### CODE REVIEW

File: rust_src/src/keymap.rs [hunk 1]
```diff
@@ -112,4 +113,31 @@ pub fn lookup_key(keymap: LispObject, key: LispObject, accept_default: LispObjec
...
+    command: LispObject,
+    mapvar: LispObject,
+    name: LispObject,
+) -> LispObject {
+    let map = unsafe { LispObject::from(Fmake_sparse_keymap(name.to_raw())) };
+    unsafe { Fset(command.to_raw(), map.to_raw()) };
+    if mapvar.is_not_nil() {
+        unsafe { Fset(mapvar.to_raw(), map.to_raw()) };
+    } else {
+        unsafe { Fset(command.to_raw(), map.to_raw()) };
```
Comment (brotzeit) [inline rust_src/src/keymap.rs]: Looks redundant to me ?

File: rust_src/src/keymap.rs [hunk 2]
```diff
@@ -112,4 +113,31 @@ pub fn lookup_key(keymap: LispObject, key: LispObject, accept_default: LispObjec
...
+/// string for the map.  This is required to use the keymap as a menu.
+/// This function returns COMMAND.
+#[lisp_fn(min = "1")]
+pub fn define_prefix_command(
+    command: LispObject,
+    mapvar: LispObject,
+    name: LispObject,
+) -> LispObject {
+    let map = unsafe { LispObject::from(Fmake_sparse_keymap(name.to_raw())) };
+    unsafe { Fset(command.to_raw(), map.to_raw()) };
```
Comment (DavidDeSimone) [inline rust_src/src/keymap.rs]: this was a `Ffset` before, and now it's a `Fset`

Comment (brotzeit) [reply rust_src/src/keymap.rs]: 🙈

File: rust_src/src/keymap.rs [hunk 3]
```diff
@@ -112,4 +113,30 @@ pub fn lookup_key(keymap: LispObject, key: LispObject, accept_default: LispObjec
...
+/// The third optional argument NAME, if given, supplies a menu name
+/// string for the map.  This is required to use the keymap as a menu.
+/// This function returns COMMAND.
+#[lisp_fn(min = "1")]
+pub fn define_prefix_command(
+    command: LispObject,
+    mapvar: LispObject,
+    name: LispObject,
+) -> LispObject {
+    let map = unsafe { LispObject::from(Fmake_sparse_keymap(name.to_raw())) };
```
Comment (shaleh) [inline rust_src/src/keymap.rs]: Hmm. Maybe just put the whole function inside an unsafe block?

Comment (brotzeit) [reply rust_src/src/keymap.rs]: I'd like to keep it this way, as it explicitely shows which parts are unsafe. And it makes it easier to remove unneccessary unsafe blocks when porting the used functions.

**Comment by shaleh** [comment]
twiddle build

**Pull request merged**